test/testdata/rewriter/minitest_tables.rb

```diff
@@ -86,13 +86,93 @@ def self.it(name, &blk); end
   end
 
 
-  test_each [1, 2, 3] do |k, v| # error: Wrong number of parameters for `test_each` block
-    it "does not handle more than one argument" do
+  test_each [1, 2, 3] do |k, v|
+    it "blocks get padded with NilClass arguments" do
+      T.reveal_type(k) # error: Revealed type: `Integer`
+      T.reveal_type(v) # error: Revealed type: `NilClass`
+    end
+
+    it "multiple it blocks are ok with multiple arguments" do
+      T.reveal_type(k) # error: Revealed type: `Integer`
+      T.reveal_type(v) # error: Revealed type: `NilClass`
+    end
+  end
+
+  test_each [1, 2, 3] do |(k, v)|
+    it "destructured blocks get padded with NilClass arguments" do
+      T.reveal_type(k) # error: Revealed type: `Integer`
+      T.reveal_type(v) # error: Revealed type: `NilClass`
+    end
+
+    it "multiple it blocks are ok with multiple destructured arguments" do
+      T.reveal_type(k) # error: Revealed type: `Integer`
+      T.reveal_type(v) # error: Revealed type: `NilClass`
+    end
+  end
+
+  test_each [[1, ['hi', false]], [2, ['bye', true]]] do |i, (s,b), (x, y)|
+    it "handles mixed destructuring and positional arguments" do
+      T.reveal_type(i) # error: Revealed type: `Integer`
+      T.reveal_type(s) # error: Revealed type: `String`
+      T.reveal_type(b) # error: Revealed type: `T::Boolean`
+      T.reveal_type(x) # error: Revealed type: `NilClass`
+      T.reveal_type(y) # error: Revealed type: `NilClass`
+    end
+
+    it "multiple it blocks are ok with mixed argument styles" do
+      T.reveal_type(i) # error: Revealed type: `Integer`
+      T.reveal_type(s) # error: Revealed type: `String`
+      T.reveal_type(b) # error: Revealed type: `T::Boolean`
+    end
+  end
+
+  CONST_LIST_TUPLE = [[1,'a'], [2,'b']]
+  test_each CONST_LIST_TUPLE do |i, s|
+    it "succeeds with a constant list of tuples" do
+      T.reveal_type(i) # error: type: `T.untyped`
+      T.reveal_type(s) # error: type: `T.untyped`
+    end
+  end
+
+  test_each CONST_LIST_TUPLE do |(i, s)|
+    it "succeeds with a constant list of tuples and destructuring" do
+      T.reveal_type(i) # error: type: `T.untyped`
+      T.reveal_type(s) # error: type: `T.untyped`
+    end
+  end
+
+  ANOTHER_CONST_LIST_TUPLE = T.let([[1,'a'], [2,'b']], T::Array[[Integer, String]])
+  test_each ANOTHER_CONST_LIST_TUPLE do |i, s|
+    it "succeeds with a typed constant tuple list" do
+      T.reveal_type(i) # error: type: `Integer`
+      T.reveal_type(s) # error: type: `String`
+    end
+  end
+
+  test_each ANOTHER_CONST_LIST_TUPLE do |(i, s)|
+    it "succeeds with a typed constant tuple list with destructuring" do
+      T.reveal_type(i) # error: type: `Integer`
+      T.reveal_type(s) # error: type: `String`
+    end
+  end
+
+  local_tuple = [[1,'a'], [2,'b']]
+  test_each local_tuple do |i,s|
+    it "succeed with local variables but cannot type them" do
+      T.reveal_type(i) # error: type: `T.untyped`
+      T.reveal_type(s) # error: type: `T.untyped`
+    end
+  end
```

Comment:
Can you add a test for when the programmer does the destructuring manually?

```ruby

test_each([[1, 2], [3, 4]]) do |pair|
  x, y = pair
  it 'manual destructuring' do
    # ...
  end
end
```

(no preference whether this should work or not, just want to capture the behavior)

Reply:
Happy to add the test, it's going to fail currently as we only allow `it` blocks within `test_each`.